rebasehelper/specfile.py

```diff
@@ -1114,6 +1114,21 @@ def get_prep_section(self):
                 result.append(prep.pop(0))
         return result
 
+    @staticmethod
+    def get_subpackage_name(files_section):
+        """Gets subpackage name based on the %files section."""
+        parser = SilentArgumentParser()
+        parser.add_argument('-n', default=None)
+        parser.add_argument('-f')
+        parser.add_argument('subpackage', nargs='?', default=None)
```

Comment:
The same argument parser is used in `SpecFile.get_main_files_section()`. I think it would be better to merge the two methods, or call this one in `SpecFile.get_main_files_section()`.

Also, the `-n` option should be `store_true` action, it just affects how `subpackage` argument is treated.

Reply:
Should it really be `store_true`? The value of the argument is returned as the full subpackage name so isn't it necessary to use it as a regular argument with `nargs=1`?

Reply:
But that's not a value of the `-n` option, rather it's `subpackage` argument.

Reply:
I just tested it and I don't think that RPM implements it this way. Tested on package `pidgin`:

With `%files -f pidgin.lang -n libpurple` the build passed but after swapping the arguments to `%files -n -f pidgin.lang libpurple`, building SRPM failed with error message `Too many names`.  Based on this fact, I assume that the `-n` option doesn't work as a switch.

Reply:
Ok, I stand corrected.